### ISSUE TITLE
RevelJS: How to create a horizontal line between slide content and footnotes

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#8545]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/8534

````markdown
---
title: test
format:
  revealjs:
    center: true
reference-location: section
---

## Slide 1

- bullet one
- bullet two^[first footnote]
- This is the third bullet

## Slide 2

more test text here^[second footnote]
````

This leads to wrong numbering
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/d5b44f69-a5db-4011-8b27-1d1920b71292)

Either this is an issue, or we should just disallow `reference-location: section` as I am not sure it makes sense in revealjs context

**Comment by mcanouil** [comment]
> Either this is an issue, or we should just disallow reference-location: section as I am not sure it makes sense in revealjs context

It does number correctly, but from level one header.

**Comment by cderv** [comment]
Sorry, I don't understand. In HTML the numbering of footnote is sequential and by default at the end. the `reference-location: section` just moves under each section but keeps the sequential numbering in the doc.  

````markdown
---
title: test
format: html
reference-location: section
---

# Header 1 

## Sub 1

- bullet one
- bullet two^[first footnote]
- This is the third bullet

## Sub 2

more test text here^[second footnote]

# Header 2 

## Sub 3

more test text here^[second footnote]

**Comment by mcanouil** [comment]
Never mind, the numbering is completely messed up with the option regardless of headings.
<img width="770" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/9e83911b-f3e5-44be-ac90-0a99cde30d00">

**Comment by cderv** [comment]
yes, this is why I opened the issue in the first place.

**Comment by mcanouil** [comment]
It's messed up but only in the footer.

FYI `reference-location: document`
<img width="906" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/c5a8c366-3600-45f2-897f-e7105190bbe9">

**Comment by cderv** [comment]
Yes, as I said above in https://github.com/quarto-dev/quarto-cli/issues/8545#issuecomment-1922188181 we do a special processing in `format-reveal.ts` when `reference-location` is different than document. 

This process is about getting the footnotes and splitting them on each related slides.
